File: JS/8kyu/square&sum.js
// https://www.codewars.com/kata/515e271a311df0350d00000f/javascript

function squareSum(numbers) {
    return numbers.reduce((r, e) => {
        r += Math.pow(e, 2)
        return r
    }, 0)
}

console.log(squareSum([0, 3, 4, 5]))

File: C#/8kyu/square&sum.cs
// https://www.codewars.com/kata/515e271a311df0350d00000f/javascript

using System;
using System.Linq;
public static class Kata
{
    public static int SquareSum(int[] n)
    {
        return n.Select(e => e * e).Sum();
    }
}

// // Better solution

public static class Kata
{
    public static int SquareSum(int[] n)
    {
        return n.Sum(x => x * x);
    }
}